p2p/libp2p/libp2pConnectionMonitor_test.go

```diff
@@ -60,3 +62,84 @@ func TestNewLibp2pConnectionMonitor_OnDisconnectedUnderThresholdShouldCallReconn
 		assert.Fail(t, "timeout waiting to call reconnect")
 	}
 }
+
+func TestNewLibp2pConnectionMonitor_DefaultTriming(t *testing.T) {
+	t.Parallel()
+
+	cm := newLibp2pConnectionMonitor(nil, 0)
+
+	assert.NotNil(t, cm)
+	assert.Equal(t, 0, cm.thresholdDiscoveryResume)
+	assert.Equal(t, math.MaxInt32, cm.thresholdDiscoveryPause)
+	assert.Equal(t, math.MaxInt32, cm.thresholdConnTrim)
+}
+
+func TestNewLibp2pConnectionMonitor_Triming(t *testing.T) {
+	t.Parallel()
+
+	pauseCallCount := 0
+	resumeCallCount := 0
+
+	rc := mock.ReconnecterStub{
+		ReconnectToNetworkCalled: func() <-chan struct{} {
+			ch := make(chan struct{})
+			defer func() { ch <- struct{}{} }()
+			return ch
+		},
+		PauseCall:  func() { pauseCallCount++ },
+		ResumeCall: func() { resumeCallCount++ },
+	}
+
+	cm := newLibp2pConnectionMonitor(&rc, 10)
+
+	assert.NotNil(t, cm)
+	assert.Equal(t, 8, cm.thresholdDiscoveryResume)
+	assert.Equal(t, 10, cm.thresholdDiscoveryPause)
+	assert.Equal(t, 12, cm.thresholdConnTrim)
+
+	netFact := func(cnt int) network.Network {
+		cntr := cnt
+		currentCount := &cntr
+		return &mock.NetworkStub{
+			ConnsCalled: func() []network.Conn {
+				return make([]network.Conn, *currentCount)
+			},
+
+			PeersCall: func() []peer.ID {
+				return make([]peer.ID, *currentCount)
+			},
+
+			ClosePeerCall: func(peer.ID) error {
+				*currentCount--
+				return nil
+			},
+		}
+	}
+
+	assert.Equal(t, 0, pauseCallCount)
+	assert.Equal(t, 0, resumeCallCount)
+
+	cm.Connected(netFact(5), nil)
```

Comment:
We could have used a map for these kind of tests. Something like map[int][]int there the map key is the parameter sent to netFact function. The slice would have contained the expected values passed to assert.Equal function.